libff/algebra/fields/binary_field.hpp

```diff
@@ -0,0 +1,110 @@
+/** @file
+ *****************************************************************************
+ Declaration of common API for all finite fields in the binary/ directory.
+
+ Currently NOT used by the fields in this library. This class is not actually
+ the parent class of any field. All APIs are enforced through tests instead.
+
+ The reason for this is to ensure high performance of all fields. This class
+ exists as documentation for common API between fields.
+
+ Includes fields F2^n for specified n. All of the binary entension fields must
+ implement all functions declared in this class.
+ *****************************************************************************
+ * @author     This file is part of libff, developed by SCIPR Lab
+ *             and contributors (see AUTHORS).
+ * @copyright  MIT license (see LICENSE file)
+ *****************************************************************************/
+#include <libff/algebra/field_utils/bigint.hpp>
+#include <vector>
+
+namespace libff {
+
+template<typename T>
+class BinaryField;
+
+/* The type parameter T is intended to be set to the child class
+ * when this class is extended. For example,
+ * class gf32 : public BinaryField<gf32> ...
+ */
+template<typename T>
+class BinaryField {
+public:
+    /* Functions unique to binary fields */
+
+    static const constexpr uint64_t modulus_;
```

Comment:
TODO: We should eventaully add some documentation about how modulii are represented, no need to block this PR though!